src/support/generation.c

```diff
@@ -109,7 +106,15 @@ void
 			/* Ensure we only read the value once. */
 			WT_ORDERED_READ(v, s->generations[which]);
 
-			if (v == 0 || generation <= v)
+			/*
+			 * The generation argument is newer than the limit. Wait
+			 * for threads in generations older than the argument
+			 * generation, threads in argument generations are OK.
+			 *
+			 * The thread's generation may be 0 (that is, not set),
+			 * and that's always less than the argument generation.
+			 */
+			if (v < generation)
 				break;
```

Comment:
This seems backwards to me: when we see a session in an older generation than the one we're waiting on, we move on to the next session?  Assuming threads only stay in a generation temporarily, we'd expect to eventually see `v == 0`, but otherwise I think we should only move on to the next session when `v > generation`.